x-pack/test/idp-fixture/src/main/java/org/elasticsearch/test/fixtures/idp/OidcProviderTestContainer.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.test.fixtures.idp;
+
+import org.elasticsearch.test.fixtures.CacheableTestFixture;
+import org.elasticsearch.test.fixtures.testcontainers.DockerEnvironmentAwareTestContainer;
+import org.junit.rules.TestRule;
+import org.testcontainers.containers.Network;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.images.builder.Transferable;
+
+public final class OidcProviderTestContainer extends DockerEnvironmentAwareTestContainer implements TestRule, CacheableTestFixture {
+
+    private static final int PORT = 8080;
+
+    public OidcProviderTestContainer(Network network) {
+        super(
+            new ImageFromDockerfile("es-oidc-provider-fixture", false).withFileFromClasspath("oidc/setup.sh", "/oidc/setup.sh")
+                // we cannot make use of docker file builder
+                // as it does not support multi-stage builds
+                .withFileFromClasspath("Dockerfile", "oidc/Dockerfile")
+        );
+        withNetworkAliases("oidc-provider");
+        withNetwork(network);
+        addExposedPort(PORT);
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        copyFileToContainer(
```

Comment:
Is there a reason we don't do this in an entrypoint script in the container?

Reply:
As I understand it I would need access to the mapped port at that point. is that supported by an ENTRYPOINT to have access to that. I think it doesn't work how testcontainers are wired up.

Reply:
Hmm, yeah, that makes sense. I thought maybe this would be exposed in the container as an environment variable but seems like no such mechanism exists.